MSVS2017 build warning: G-API sample (privacy_masking_camera)

[Build](http://pullrequest.opencv.org/buildbot/builders/master-win64-vc15/builds/283):

```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include\algorithm(1443): warning C4244: '=': conversion from 'int' to 'char', possible loss of data [C:\build\master-win64-vc15\build\modules\gapi\example_gapi_privacy_masking_camera.vcxproj]
  C:\build\master-win64-vc15\opencv\modules\gapi\samples\privacy_masking_camera.cpp(38): note: see reference to function template instantiation '_OutIt std::transform<std::_String_iterator<std::_String_val<std::_Simple_types<_Ty>>>,std::_String_iterator<std::_String_val<std::_Simple_types<_Ty>>>,`anonymous-namespace'::weights_path::<lambda_b3c87623445e81d079f66bc43b609a03>>(const _InIt,const _InIt,_OutIt,_Fn)' being compiled
          with
          [
              _OutIt=std::_String_iterator<std::_String_val<std::_Simple_types<char>>>,
              _Ty=char,
              _InIt=std::_String_iterator<std::_String_val<std::_Simple_types<char>>>,
              _Fn=`anonymous-namespace'::weights_path::<lambda_b3c87623445e81d079f66bc43b609a03>
          ]
```

--- a/modules/gapi/samples/privacy_masking_camera.cpp
+++ b/modules/gapi/samples/privacy_masking_camera.cpp
@@ -35,7 +35,7 @@ std::string weights_path(const std::string &model_path) {
 
     auto ext = model_path.substr(sz - EXT_LEN);
 
-    std::transform(ext.begin(), ext.end(), ext.begin(), [](unsigned char c){ return std::tolower(c); });
+    std::transform(ext.begin(), ext.end(), ext.begin(), [](unsigned char c){ return static_cast<unsigned char>(std::tolower(c)); });
     CV_Assert(ext == ".xml");
 
     return model_path.substr(0u, sz - EXT_LEN) + ".bin";

